src/main/java/com/google/edith/DealItem.java

```diff
@@ -38,34 +33,26 @@ public void setPrice(String price) {
   }
 
   /**
-    * Weight values from grocerydata.csv come in three main types:
-    * '' is invalid type, 'dozen' 'per' and 'head' are salvageable
-    * types, as their meaning can be interpreted, and any input that
-    * starts with a number and ends with a weight unit is valid type.
-    *
-    * ex: Inputs of 'per' or 'head' indicate that the price given is
-    *     for one unit of that item. Therefore, setting the weight to
-    *     one would let the value algorithm still work.
-    *     'per lb' -> this.weight = 1.0
-    * ex: Input weight '64 fl oz' -> this.weight = 64.0
-    */
+   * Weight values from grocerydata.csv come in three main types: '' is invalid type, 'dozen' 'per'
+   * and 'head' are salvageable types, as their meaning can be interpreted, and any input that
+   * starts with a number and ends with a weight unit is valid type.
+   *
+   * <p>ex: Inputs of 'per' or 'head' indicate that the price given is for one unit of that item.
+   * Therefore, setting the weight to one would let the value algorithm still work. 'per lb' ->
+   * this.weight = 1.0 ex: Input weight '64 fl oz' -> this.weight = 64.0
+   */
   public void setWeight(String weight) {
     if (weight.isEmpty()) {
       this.weight = 0.0;
-    }
-    else if (weight.equals("dozen")) {
+    } else if (weight.equals("dozen")) {
       this.weight = 12.0;
-    }
-    else if (weight.length() == 2) {
+    } else if (weight.length() == 2) {
```

Comment:
we also might be able to combine this check with the line 45 check. it might be a bit strange reading isEmpty and ==2 but if it helps you can leave a comment. Your call just a thought

`if (weight.isEmpty() || weight.length() == 2) {`